kore/src/Kore/Internal/MultiOr.hs

```diff
@@ -139,56 +146,17 @@ instance
     {-# INLINE rename #-}
 
 bottom :: MultiOr term
-bottom = MultiOr []
-
-{- | 'OrBool' is an some sort of Bool data type used when evaluating things
-inside a 'MultiOr'.
--}
-data OrBool = OrTrue | OrFalse | OrUnknown
-
-{- | Simplify the disjunction.
-
-The arguments are simplified by filtering on @\\top@ and @\\bottom@. The
-idempotency property of disjunction (@\\or(φ,φ)=φ@) is applied to remove
-duplicated items from the result.
-
-See also: 'filterUnique'
--}
-filterOr ::
-    (Ord term, TopBottom term) =>
-    MultiOr term ->
-    MultiOr term
-filterOr = filterGeneric patternToOrBool . filterUnique
-
-{- | Simplify the disjunction by eliminating duplicate elements.
-
-The idempotency property of disjunction (@\\or(φ,φ)=φ@) is applied to remove
-duplicated items from the result.
-
-Note: Items are compared with their Ord instance. This does not attempt
-to account separately for things like α-equivalence, so, if that is not
-included in the Ord instance, items containing @\\forall@ and
-@\\exists@ may be considered inequal although they are equivalent in
-a logical sense.
--}
-filterUnique :: Ord a => MultiOr a -> MultiOr a
-filterUnique = MultiOr . Set.toList . Set.fromList . getMultiOr
-
--- | 'make' constructs a normalized 'MultiOr'.
-make ::
-    (Ord term, TopBottom term) =>
-    [term] ->
-    MultiOr term
-make patts = filterOr (MultiOr patts)
+bottom = MultiOrBottom
 
 -- | Construct a normalized 'MultiOr' from a single pattern.
 singleton ::
     TopBottom term =>
     term ->
     MultiOr term
 singleton term
-    | isBottom term = MultiOr []
-    | otherwise = MultiOr [term]
+    | isBottom term = MultiOrBottom
```

Comment:
Why not `make [term]`, like the implementation in MultiAnd?

Reply:
We avoid a requirement on `Ord term`.... I have unfolded the singleton function in MultiAnd for symmetry if thats ok?